admin/user.go

```diff
@@ -20,6 +21,21 @@ func (s *Service) CreateOrUpdateUser(ctx context.Context, email, name, photoURL
 	// Update user if exists
 	user, err := s.DB.FindUserByEmail(ctx, email)
 	if err == nil {
+		// get existing orgs for the user
+		existingOrgMemberships := make(map[string]bool)
+		orgs, err := s.DB.FindOrganizationsForUser(ctx, user.ID, "", math.MaxInt)
+		if err != nil {
+			return nil, err
+		}
+		for _, org := range orgs {
+			existingOrgMemberships[org.ID] = true
+		}
+
+		err = s.handleAutoinvites(ctx, user, existingOrgMemberships)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
We shouldn't wait for user to log in again before adding existing users. Can just keep the logic here the same. Instead, when setting the whitelist, it should scan for and add any matching existing users to the org in the API handler.

Reply:
This sounds expensive if there are a lot of users.

Reply:
It's the product requirement – I guess the idea is that if you add a whitelisted domain, you expect member listings to reflect everyone with access right away. 

I don't think the performance is that bad, though. This is an API that will get hit very rarely, even at scale (will probably never be called more than a few times per hour), and we're not going to have *that* many users on a single domain anytime soon.

If performance ever becomes a problem, we could move it to a background job, but I think it's safe to do in the API handler for now.